src/pages/_app.tsx

```diff
@@ -83,6 +83,22 @@ function MyApp({ Component, pageProps }: any) {
     window.history.replaceState(null, '', url.href);
   }
 
+  useEffect(() => {
+    const handleComplete = (url: string) => {
+      if (url === router.asPath) {
+        document.body.style.cursor = 'auto';
+      }
+    };
+
+    router.events.on('routeChangeComplete', handleComplete);
+    router.events.on('routeChangeError', handleComplete);
+
+    return () => {
+      router.events.off('routeChangeComplete', handleComplete);
+      router.events.off('routeChangeError', handleComplete);
+    };
+  }, [router]);
```

Comment:
i dont think there is a need for this since this is being used in the `pages/search/index.tsx` file as well right?

Reply:
its needed here too
coz when redirected away from `/search`
it'll show loading cursor constantly

Reply:
we have set a loading cursor while the search is happening 
or to be exact when the url changes. 

this means, the routeChange listeners will be activated when the url changes, even if it is query parameters, this will be called, and show the loading cursor

```ts
        document.body.style.cursor = 'wait';
```

so when moved away from `/search?q=eaf` to lets say `/bounties`
the router listener `routeChangeStart` in `/search` is activated and starts the loading cursor
but now that we have unmounted `/search` the router listener `routeChangeComplete` in `/search` is never called
so the loading cursor continues to show even after route and is never unset.

so to unset that, i've added a `routerChangeComplete` listener on `_app.tsx` as well

Reply:
nvm, found a cleaner way to do so.

https://github.com/SuperteamDAO/earn/pull/547/commits/9dcc06b7fdc263fafe565413caaa9161da69218d?diff=split&w=0